File: tarai/tarai.c
#include "../lib/bench.c"

int tarai(int x, int y, int z){
	if(x <= y){
		return y;
	}else{
		return tarai(tarai(x-1,y,z),tarai(y-1,z,x),tarai(z-1,x,y));
	}
}

int main(){
	bench_start();
	int result = tarai(12,6,0);
	bench_end();
}

File: tarai/tarai.rb
require_relative '../lib/bench.rb'

def tarai(x, y, z)
	if x <= y
		y
	else
		tarai(tarai(x-1,y,z),tarai(y-1,z,x),tarai(z-1,x,y))
	end
end


Bench.bench_start()
result = tarai(12,6,0)
Bench.bench_end()